enterprise/suppress-user/factory.go

```diff
@@ -26,57 +31,197 @@ func (m *Factory) Setup(ctx context.Context, backendConfig backendconfig.Backend
 
-		repository, err = NewBadgerRepository(
-			path,
-			m.Log,
-			WithSeederSource(seederSource),
-			WithMaxSeedWait(config.GetDuration("BackendConfig.Regulations.maxSeedWait", 5, time.Second)))
-		if err != nil {
-			return nil, fmt.Errorf("could not create badger repository: %w", err)
+			subCtx, latestSyncCancel := context.WithCancel(ctx)
+			rruntime.Go(func() {
+				latestSyncer.SyncLoop(subCtx)
+				err = latestRepo.Stop()
+				if err != nil {
+					m.Log.Error("Latest Sync failed: could not stop repo: %w", err)
+				}
+				err = os.RemoveAll(latestSuppressionPath)
+				if err != nil {
+					m.Log.Error("Latest Sync failed: could not remove repo: %w", err)
+				}
+			})
+			repo.Repository = latestRepo
+			rruntime.Go(func() {
+				var fullSyncer *Syncer
+				var fullRepo Repository
+				var err error
+
+				m.retry(ctx,
+					func() error {
+						fullSyncer, fullRepo, err = m.newSyncerWithBadgerRepo(fullSuppressionPath, fullDataSeed, 0, identity, pollInterval)
+						return err
+					}, 5*time.Second)
+
+				m.retry(ctx,
+					func() error { return fullSyncer.Sync(ctx) },
+					5*time.Second)
+				_, err = os.Create(filepath.Join(fullSuppressionPath, model.SyncDoneMarker))
+				if err != nil {
+					m.Log.Error("Could not create sync done marker: %w", err)
+				}
+				repo.Switch(fullRepo)
+				latestSyncCancel()
+				fullSyncer.SyncLoop(ctx)
+				_ = fullRepo.Stop()
+			})
+			return newHandler(repo, m.Log), nil
+		} else {
+			syncer, fullRepo, err := m.newSyncerWithBadgerRepo(fullSuppressionPath, nil, 0, identity, pollInterval)
+			if err != nil {
+				return nil, err
+			}
+			rruntime.Go(func() {
+				syncer.SyncLoop(ctx)
+				_ = fullRepo.Stop()
+			})
+			return newHandler(fullRepo, m.Log), nil
 		}
 	} else {
-		repository = NewMemoryRepository(m.Log)
+		memoryRepo := NewMemoryRepository(m.Log)
+		syncer, err := NewSyncer(
+			config.GetString("SUPPRESS_USER_BACKEND_URL", "https://api.rudderstack.com"),
+			backendConfig.Identity(),
+			memoryRepo,
+			WithLogger(m.Log),
+			WithHttpClient(&http.Client{Timeout: config.GetDuration("HttpClient.suppressUser.timeout", 30, time.Second)}),
+			WithPageSize(config.GetInt("BackendConfig.Regulations.pageSize", 5000)),
+			WithPollIntervalFn(func() time.Duration { return pollInterval }),
+		)
+		if err != nil {
+			return nil, err
+		}
+		rruntime.Go(func() {
+			syncer.SyncLoop(ctx)
+			err = memoryRepo.Stop()
+		})
+		h := newHandler(memoryRepo, m.Log)
+
+		return h, nil
 	}
+}
 
-	var pollInterval time.Duration
-	config.RegisterDurationConfigVariable(300, &pollInterval, true, time.Second, "BackendConfig.Regulations.pollInterval")
+func alreadySeeded(repoPath string) bool {
+	_, err := os.Stat(repoPath)
+	if os.IsNotExist(err) {
+		return false
+	}
+	_, err = os.Stat(path.Join(repoPath, model.SyncDoneMarker))
+	if os.IsNotExist(err) {
+		os.RemoveAll(repoPath)
+		return false
+	}
+	return true
+}
 
+func (m *Factory) retry(ctx context.Context, f func() error, wait time.Duration) {
```

Comment:
minor: it might be better to name it `retryIndefinitely`
```suggestion
func (m *Factory) retryIndefinitely(ctx context.Context, f func() error, wait time.Duration) {
```